### PR TITLE
Make MediaPlayer.prepare() a sync call

### PR DESCRIPTION
**Pull request by junmin-kim**
The caller will be blocked until prepare is done.

### CODE REVIEW

File: apps/examples/cxxtest/MediaPlayer.cxx [hunk 1]
```diff
@@ -113,12 +117,16 @@ namespace Media
 
 	void MediaPlayer::_prepare()
 	{
-		curState = PLAYER_STATE_READY;
+		unique_lock<mutex> lock(*cMtx);
```
Comment (jsdosa) [inline apps/examples/cxxtest/MediaPlayer.cxx]: @junmin-kim why delete state~?

File: apps/examples/cxxtest/MediaPlayer.cxx [hunk 2]
```diff
@@ -113,12 +117,16 @@ namespace Media
...
+		std::cout << "Player prepare" << std::endl;
+		cvSync.notify_one();
 	}
 
 	void MediaPlayer::_unprepare()
 	{
-		curState = PLAYER_STATE_IDLE;
+		unique_lock<mutex> lock(*cMtx);
```
Comment (jsdosa) [inline apps/examples/cxxtest/MediaPlayer.cxx]: @junmin-kim why delete state~?

File: apps/examples/cxxtest/MediaPlayer.cxx (hunk 1 above)
Comment (junmin-kim) [reply apps/examples/cxxtest/MediaPlayer.cxx]: It will set state at prepare, not _prepare.
Because it is sync call.

Comment (jsdosa) [reply apps/examples/cxxtest/MediaPlayer.cxx]: So when will it be ready?

Comment (junmin-kim) [reply apps/examples/cxxtest/MediaPlayer.cxx]: At the end of prepare.

```
	player_result_t MediaPlayer::prepare() // sync call
	{
		unique_lock<mutex> lock(*cMtx);
		enqueue([this]() {_prepare(); });
		cvSync.wait(lock);

		curState = PLAYER_STATE_READY;
		return PLAYER_OK;
	}
```

Comment (jsdosa) [reply apps/examples/cxxtest/MediaPlayer.cxx]: yes yes ~

**Pull request merged**